### PR TITLE
feat(developer):  scaffolding for Keyman Developer for VSCode🗼 

### PR DESCRIPTION
**Pull request by srl295**
- add the VSCode extension directory under `developer/src/vscode-plugin` : Keyman Developer for VSCode
- no features yet, just build infrastructure 

This will become the parent commit for both #12757 and #12790

@keymanapp-test-bot skip

Fixes: #12798

### CODE REVIEW

**Comment by srl295** [comment]
meant to be ready for review once builds passed.

File: developer/src/vscode-plugin/src/test/extension.test.ts
Comment (mcdurdin) [inline developer/src/vscode-plugin/src/test/extension.test.ts:1]: should be called extension.tests.ts

File: developer/src/vscode-plugin/package.json [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    "Other"
+  ],
+  "activationEvents": [],
+  "main": "./out/extension.js",
+  "contributes": {
+  },
+  "scripts": {
+    "vscode:prepublish": "npm run compile",
+    "compile": "tsc -p ./",
+    "watch": "tsc -watch -p ./",
```
Comment (mcdurdin) [inline developer/src/vscode-plugin/package.json:22]: I notice some of the comments from previous PR haven't yet been applied here, is that intentional?

**Review by mcdurdin — APPROVED** [review]
LGTM, would be good to check the comments from #12757 as there are a few things that were noted there

File: developer/src/vscode-plugin/package.json (hunk 1 above)
Comment (srl295) [reply developer/src/vscode-plugin/package.json:22]: reviewing.

**Comment by srl295** [comment]
@mcdurdin double checked all of the comments… all in…

**Pull request merged**